layers/core_checks/cc_device_memory.cpp

```diff
@@ -1253,12 +1253,12 @@ bool CoreChecks::PreCallValidateMapMemory2KHR(VkDevice device, const VkMemoryMap
                                      pMemoryMapInfo->offset);
                 }
 
-                if (pMemoryMapInfo->size != VK_WHOLE_SIZE) {
+                if (pMemoryMapInfo->size != VK_WHOLE_SIZE && pMemoryMapInfo->size != mem_info->allocate_info.allocationSize) {
                     skip |= LogError("VUID-VkMemoryMapInfoKHR-flags-09572", pMemoryMapInfo->memory, info_loc.dot(Field::size),
                                      "(0x%" PRIx64
-                                     ") is not VK_WHOLE_SIZE while VK_MEMORY_MAP_PLACED_BIT_EXT is set and "
-                                     "memoryMapRangePlaced is not enabled.",
-                                     pMemoryMapInfo->size);
+                                     ") is not VK_WHOLE_SIZE or the size of memory (%" PRIu64
```

Comment:
```suggestion
                                     ") is not VK_WHOLE_SIZE and the size of memory (%" PRIu64
```

Am I wrong here, this wording doesn't match the logic above

Reply:
Ugh... english. The check is "is not foo and is not bar" while the text here is "is not foo or bar". I think the english means the same but if you've got a suggestion for how to phrase it better, I'm all ears.